jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/PathMappings.java

```diff
@@ -231,24 +295,48 @@ public boolean put(PathSpec pathSpec, E resource)
         switch (pathSpec.getGroup())
         {
             case EXACT:
-                String exact = pathSpec.getPrefix();
-                while (exact != null && !_exactMap.put(exact, entry))
+                if (pathSpec instanceof ServletPathSpec)
+                {
+                    String exact = pathSpec.getDeclaration();
```

Comment:
I know it is not part of this PR, but can you put a comment in the while below that says what the loop is for (i.e. it is growing the capacity of the trie).  Took me a few minutes to work that out.

Reply:
Done.